vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -9,15 +9,15 @@ import { IEventStream } from '../EventStream/EventStream';
 import { WebRequestError } from '../EventStream/EventStreamEvents';
 
 export class WebRequestWorker {
-    private cachedData: any;
-    private currentRequest: Promise<any> | undefined;
+    private cachedData: string | undefined;
+    private currentRequest: Promise<string | undefined> | undefined;
 
     constructor(private readonly extensionState: Memento,
                 private readonly eventStream: IEventStream,
                 private readonly uri: string,
                 private readonly extensionStateKey: string) {}
 
-    public async getCachedData(): Promise<string> {
+    public async getCachedData(): Promise<string | undefined> {

```

Comment:
Why is adding the undefined here necessary?

Reply:
The `this.extensionState.get<string>` returns the union `string | undefined`, making `this.cachedData` a union of `string | undefined`, which then makes the entire method that too.